src/Microsoft.AspNetCore.SignalR.Core/IHubProtocolResolver.cs

```diff
@@ -6,9 +6,22 @@
 
 namespace Microsoft.AspNetCore.SignalR
 {
+    /// <summary>
+    /// A resolver abstraction for working with <see cref="IHubProtocol"/> instances.
+    /// </summary>
     public interface IHubProtocolResolver
     {
+        /// <summary>
+        /// Gets a collection of all protocols.
+        /// </summary>
         IReadOnlyList<IHubProtocol> AllProtocols { get; }
+
+        /// <summary>
+        /// Gets a protocol using the specified protocol name and a collection of supported protocols.
```

Comment:
Gets the hub protocol with the specified name, if it is allowed by the specified list of supported protocols.